src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1338,12 +1341,13 @@ trait RuntimeCosting extends CostingRules with DataCosting with Slicing { IR: Ev
         }
 
       case Values.Tuple(InSeq(Seq(x, y))) =>
-        RCCostedPair(x, y, opCost(Seq(x.cost, y.cost), CostTable.newPairValueCost))
+        RCCostedPair(x, y, opCost(x.value, Seq(x.cost, y.cost), CostTable.newPairValueCost))
```

Comment:
The same here, try `Pair(x.value, y.value)` in opCost first argument

Reply:
Fixed.